### DISCUSSION TITLE
Suggested / historical D. and D.R display formats.

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#129] [Q&A] [answered]
I've not played with any other Forth in a long long while and even then I never got around to the Double Word set.  What is the suggested (historical) output for `D.` and `D.R`.  I know they are implementation defined free format fields, but I'm curious as to what has gone before.

Does one display the high and low words separately (easy to recognise and re-input at the prompt) or "combined" to display a single large number (which seems messy to implement).

```
ok : D. DABS . U. ;
ok $dead $beef HEX D.
BEEF DEAD ok
```
or more like `bc(1)`:

```
ok $dead $beef DECIMAL D.
901658403578849173495469 ok    \ FYI 64-bit CPU
```

**Comment by MitchBradley** [comment]
combined

**Comment by MitchBradley** [comment, marked as answer]
The typical implementation is is terms of <#, i.e. `: d. ( d -- ) tuck dabs <# #s rot sign #> type space ;`.  #s is in turn a loop over #, which is often implemented via the factor `: mu/mod  (s d# n1 -- rem d#quot )    >r  0  r@  um/mod  r>  swap  >r  um/mod  r> ;`

**Reply by SirWumpus** [reply]
Thank you.  I tried that out and worked like a charm.
